### PR TITLE
Cleaner TypeListCat&SlowTypeList, + TypeListUnion.

### PR DESCRIPTION
**Pull request by dkorolev**
image

### CODE REVIEW

File: Bricks/template/typelist.h [hunk 1]
```diff
@@ -89,9 +88,99 @@ static_assert(TypeListContains<TypeListImpl<char, int, double>, char>::value, ""
...
+static_assert(std::is_same<TypeListImpl<int>, TypeListCat<TypeListImpl<int>, TypeListImpl<>>>::value, "");
+static_assert(std::is_same<TypeListImpl<int>, TypeListCat<TypeListImpl<>, TypeListImpl<int>>>::value, "");
+static_assert(
+    std::is_same<TypeListImpl<int, char, double>, TypeListCat<TypeListImpl<int>, TypeListImpl<char, double>>>::value,
+    "");
+static_assert(
+    std::is_same<TypeListImpl<int, char, double>, TypeListCat<TypeListImpl<int, char>, TypeListImpl<double>>>::value,
+    "");
+
+// `TypeListUnion` creates a `TypeList<{common part of LHS.. and RHS...}>`. It's a `TypeListCat` that allows duplicates.
```
Comment (mzhurovich) [inline Bricks/template/typelist.h]: `common part of ...` means intersection, not union, doesn't it?

File: Bricks/template/typelist.h [hunk 2]
```diff
@@ -89,9 +88,99 @@ static_assert(TypeListContains<TypeListImpl<char, int, double>, char>::value, ""
...
+static_assert(TypeListContains<TypeListImpl<char, int>, char>::value, "");
+static_assert(TypeListContains<TypeListImpl<int, char>, char>::value, "");
+
+// `TypeListCat` creates a `TypeList<LHS..., RHS...>` given `TypeList<LHS...>` and `TypeList<RHS...>`.
+// By design, it would fail at compile tim if `LHS...` and `RHS...` have at least one shared type.
+template <typename LHS, typename RHS>
+struct TypeListCatImpl;
+
+template <typename... LHS, typename... RHS>
+struct TypeListCatImpl<TypeListImpl<LHS...>, TypeListImpl<RHS...>> : DuplicateTypeInTypeList<LHS>...,
```
Comment (mzhurovich) [inline Bricks/template/typelist.h]: I'd suggest to generalize it for any number of typelists.

File: Bricks/template/typelist.h (hunk 1 above)
Comment (dkorolev) [inline Bricks/template/typelist.h]: :+1: 35731de

**Comment by mzhurovich** [comment]
LGTM!

**Pull request merged**